tests/linearizability/failpoints.go

```diff
@@ -99,13 +103,92 @@ func (f killFailpoint) Trigger(t *testing.T, ctx context.Context, clus *e2e.Etcd
 		}
 	}
 
-	err := member.Start(ctx)
+	// get endpoints excluding the killed member client URLs
+	endpoints := make([]string, 0, len(clus.EndpointsV3()))
+	for _, ed := range clus.EndpointsV3() {
+		if ed != member.EndpointsV3()[0] {
+			endpoints = append(endpoints, ed)
+		}
+	}
+
+	if f.waitForSnapshot {
+		// wait for potential leader election
+		time.Sleep(2 * time.Second)
+		cc, err := clientv3.New(clientv3.Config{
+			Endpoints:            endpoints,
+			Logger:               zap.NewNop(),
+			DialKeepAliveTime:    1 * time.Millisecond,
+			DialKeepAliveTimeout: 5 * time.Millisecond,
+			Username:             rootUserName,
+			Password:             rootUserPassword,
+		})
+		require.NoError(t, err)
+		require.NoError(t, addTestUserAuth(ctx, cc))
+		// 5000 is the default number of snapshot count catch up entries
+		// DefaultSnapshotCatchUpEntries = 5000
+		// Try manipulating e2e etcd test binary to 100 locally and snapshot-count here to speed up test
+		require.NoError(t, waitForSnapshot(ctx, cc, 5000, lg))
```

Comment:
are you trying to hit `s.authStore.Recover` etcdserver/server.go:1042? If yes, I think you need to bring back killed member, so it will actually apply snapshot.  Also you need to set snapshot-count to lower number. See https://github.com/etcd-io/etcd/issues/10219 for an example.

FYI, seams we are both working on reproducing `apply snapshot` path. My logic for https://github.com/etcd-io/etcd/issues/14726 looks similar, except I use goPanicFailpoint.

Reply:
Yeah, line 134 is restart the member.